### DISCUSSION TITLE
[과목 1] 제 1장 데이터 모델링의 이해

### DISCUSSION THREAD

**Discussion by beolggoolosori** [flataex/sqlp-study#4] [Custom 연습문제] [answered]
> **1번 문제**

다음은 데이터 모델링을 할 때 유의점 중 어떤것에 해당하는가?

id | name | address | phone_number | email_1 | email_2
-- | -- | -- | -- | -- | --
1 | 김철수 | 서울시 강남구 역삼동 | 010-1234-5678 | 철수@example.com | 철수짱@example.com
2 | 이영희 | 서울시 송파구 잠실동 | 010-5678-1234 | 영희@example.com | null
 
① 중복

② 비유연성

③ 비일관성

<br/><br/>
> **2번 문제**

**_1번 문제_**   정답의 유의점을 고려하여 새 구조로 데이터 모델링 하시오.
<br/><br/>

> **3번 문제**

아래 테이블에서 인조식별자와 보조식별자의 속성을 적으시오.

**employee**
id | name | department_id| email | system_generated_id
-- | -- | -- | -- | --
101 | 김종훈| 01| jhkim@flataex.com | 101김종훈
102 | 박지은| 01 | jepark@flataex.com  | 102박지은
103 | 권도영| 02| dykwon@flataex.com | 103권도영
105 | 이진묵| 03 | jmlee@flataex.com | 105이진묵

**department**

id | department_name
-- | -- 
01| 백오피스
02| AML 
03 | CS

정답 :

**Comment by Psjepark** [comment]
선생님 채점해주세요 👍  
---
### 1. 
2번

---
### 2.

#### User테이블 
id | name | address | phone_number | email_id
-- | -- | -- | -- | -- | 
1 | 김철수 | 서울시 강남구 역삼동 | 010-1234-5678 | 1 | 
2 | 이영희 | 서울시 송파구 잠실동 | 010-5678-1234 | 2 |


#### Email 테이블

id | email_1 | email_2
-- | -- | -- | 
1 | 철수@example.com | 철수짱@example.com
2 | 영희@example.com | null

---
### 3.
 인조식별자 : id , 보조식별자 : department_id

**Comment by white-gyu** [comment]
> **1번 문제**

다음은 데이터 모델링을 할 때 유의점 중 어떤것에 해당하는가?

id | name | address | phone_number | email_1 | email_2
-- | -- | -- | -- | -- | --
1 | 김철수 | 서울시 강남구 역삼동 | 010-1234-5678 | 철수@example.com | 철수짱@example.com
2 | 이영희 | 서울시 송파구 잠실동 | 010-5678-1234 | 영희@example.com | null
 
① 중복

② 비유연성

③ 비일관성

> ①, 잘 모르겠네요 😅 

---

> **2번 문제**

**_1번 문제_**   정답의 유의점을 고려하여 새 구조로 데이터 모델링 하시오.

<table>
<tr>
<td>
<table>
<tr>
<td align="center" colspan="4">user 테이블</td>
</tr>
<tr>
<td align="center">id</td><td align="center">name</td><td align="center">address</td><td align="center">phone_number</td>
</tr>
<tr>
<td align="center">1</td><td align="center">김철수</td><td align="center">서울시 강남구 역삼동</td><td align="center">010-1234-5678</td>
</tr>
<tr>
<td align="center">2</td><td align="center">이영희</td><td align="center">서울시 송파구 잠실동</td><td align="center">010-5678-1234</td>
</tr>
</table>
</td>
<td>
<table>
<tr>
<td align="center" colspan="3">email 테이블</td>
</tr>
<tr>
<td align="center">id</td><td align="center">user_id</td><td align="center">email</td>
<tr>
<td align="center">1</td><td align="center">1</td><td align="center">철수@example.com</td>
</tr>
<tr>
<td align="center">2</td><td align="center">1</td><td align="center">철수짱@example.com</td>
</tr>
<tr>
<td align="center">3</td><td align="center">2</td><td align="center">영희@example.com</td>
</tr>
</table>
</td>
</tr>
</table>

---

> **3번 문제**

아래 테이블에서 인조식별자와 보조식별자의 속성을 적으시오.

**employee**
id | name | department_id| email | system_generated_id
-- | -- | -- | -- | --
101 | 김종훈| 01| jhkim@flataex.com | 101김종훈
102 | 박지은| 01 | jepark@flataex.com  | 102박지은
103 | 권도영| 02| dykwon@flataex.com | 103권도영
105 | 이진묵| 03 | jmlee@flataex.com | 105이진묵

**department**

id | department_name
-- | -- 
01| 백오피스
02| AML 
03 | CS

인조 식별자 : `employe.id`
보조 식별자 : `employee.email`, `employee.system_generated_id`

> 인조 식별자 : 업무적으로 만들어지지는 않지만 원조식별자가 복잡한 구성을 가지고 있기 때문에 인위적으로 만든 식별자 
> 보조 식별자 : 유일하게 구분 가능하지만 대표성을 가지지 못해 참조관계 연결을 못함

**Comment by beolggoolosori** [comment, marked as answer]
<br/>

**1번 문제 정답**

>② 비유연성

**2번 문제 정답**

회원ID | 이름 | 주소 | 전화번호
-- | -- | -- | --
1 | 김철수 | 서울시 강남구 역삼동 | 010-1234-5678
2 | 이영희 | 서울시 송파구 잠실동 | 010-5678-1234

ID | 회원ID | 이메일
-- | -- | --
1 | 1 | 철수@example.com
2 | 1 | 철수짱@example.com
3 | 2 | 영희@example.com

**3번 문제 정답**

`문제에 약간 오류가 있습니다... 위의 테이블 정보로는 보조식별자를 정확히 알 수 없는 것 같습니다.`

>인조식별자: system_generated_id
>보조식별자: employee.email
